### DISCUSSION TITLE
A good updated tutorial/settings for H265?

### DISCUSSION THREAD

**Discussion by ZeroCool22** [staxrip/staxrip#417] [Q&A]
Someone have or know about a good settings for StaxRip (H265)?

Thx in advance.,

**Comment by Dendraspis** [comment]
A good starting point is the ***Slow*** preset and a CRF quality value of about ***18***. If there is no compatibility issue using a 10 bit file, use profile ***Main 10*** and the output depth of ***10 bit***.
But it all depends on what you want to achieve and what your source is.

When you give us more details we can try to give you better answers.

**Reply by ZeroCool22** [reply]
Thx for reply, i will be posting a RIP i did to get feedback from you guys.

**Reply by ZeroCool22** [reply]
> 
> 
> A good starting point is the _**Slow**_ preset and a CRF quality value of about _**18**_. If there is no compatibility issue using a 10 bit file, use profile _**Main 10**_ and the output depth of _**10 bit**_.
> But it all depends on what you want to achieve and what your source is.
> 
> When you give us more details we can try to give you better answers.

I would like if you (and everyone else) can give me a feedback on this RIP i did _(source was a REMUX)_:

[H256 Test](https://www.mediafire.com/file/ufej14elkfqld8g/H265_Test.rar/file) **[MediaFire]**

**Password:** `Z^2ep%h2w6wc`

**Done with:** StaxRip-x64-2.1.7.0-Stable

**Reply by Patman86** [reply]
Can't download your test file.

**Reply by ZeroCool22** [reply]
> 
> 
> Can't download your test file.

Uploaded to another host, try now.

**Reply by Dendraspis** [reply]
> Thx for reply, i will be posting a RIP i did to get feedback from you guys.

Ok, but what are your goals? Small size? Best quality? Is it enough to get a decent result that is okay for the average/normal viewer?

The video file is sadly not useful, because you don't embed the encoder settings - which is enabled by default 😉 . So the log file would be nice to view the settings. From just viewing it's hard to give help.
![SRinfo](https://user-images.githubusercontent.com/50659978/104128613-fb22d400-5368-11eb-8995-08ed1cade0d6.JPG)

**Reply by ZeroCool22** [reply]
It's not x265, it's **H265**, x265 require a very powerful processor and take long time, i use H265 to take advantage of my GPU (1080TI).

And my goals ofc. are, **small size and good quality**.

![Screenshot_4](https://user-images.githubusercontent.com/13344308/104129482-00712600-534b-11eb-94d4-683ddff550b6.png)

![Screenshot_3](https://user-images.githubusercontent.com/13344308/104129340-d0c21e00-534a-11eb-877e-ebed19e1a40b.png)

**Reply by JJKylee** [reply]
There's no such 'encoder' as H265. You're talking about _NVEnc_ **encoder** for _H265(HEVC)_ **codec(=coding format)**. You need to distinguish them in order to avoid confusion.

BTW, this is my favorite setting for an 8-bit HEVC encoding.

![StaxRip_2 1 7 0_NVEnc_settings_8bits_20210110](https://user-images.githubusercontent.com/3157518/104130021-6f3a8900-5323-11eb-96a5-59ac160d3884.jpg)

Hardware multipass enabled, and the maximum quality preset in the constant quality mode.

And about the bit depth:
I don't know how strong the internal deblocking algorithm operates for NVEnc, so instead of adopting 10 bits, I go with 8 bits for an 8-bit source along with the hardware deband filter `--vpp-deband`, _unlike_ `x265` or `aomenc`.
Maybe you can go with 10 bits instead suited to your taste.

The remaining options are just pretty self explanatory. 😄

**Reply by Dendraspis** [reply]
> It's not x265, it's **H265**, x265 require a very powerful processor and take long time, i use H265 to take advantage of my GPU (1080TI).

Ah, you mean ***NVEnc*** - H.265 is the video compression standard.
Either a powerful CPU or time.  😁  But if you can't wait...

But I think I'm out here - I didn't use NVEnc for a long time because I didn't like the results that much. NVEnc could - when I used it - give good quality, but the the file size was too big. 

> And my goals ofc. are, **small size and good quality**.

The master challenge.  😂 
For me a good starting point would be using CQP with preset P7 (=Quality), but it's hard to say without having the source.

A little hint:  Create 2min short samples for testing and/or distributing.

**Reply by JJKylee** [reply]
> But I think I'm out here - I didn't use NVEnc for a long time because I didn't like the results that much. NVEnc could - when I used it - give good quality, but the the file size was too big.

You're right. The only reason I use NVEnc is because it saves me a lot of time with a relatively _tolerable_ quality. My graphics card is GTX 1060 6G (2 generations behind 😢), and with the settings shown above, I _usually_ get about 8x for the encoding speed with a 1080p30 8-bit AVC source. Of course, the quality is not comparable to x265, but I think it can replace x264 in many cases.

But there's a catch: like CPU encoders, NVEnc heavily depends on the architecture of its encoding chips(NVENC) and NVENC API for software making use of those chips, like NVEncC by rigaya. So **encoding quality depends on which graphics card you own**. There's a table explaining what features each graphics card supports depending on its NVENC architecture: [here](https://developer.nvidia.com/video-encode-and-decode-gpu-support-matrix-new).

The most notable feature there is whether **B-frame encoding** is supported in **HEVC**. As you can see, only **Turing** and **Ampere** support this feature: **GeForce GTX 1650 GDDR6 and newer** lines. The usual GTX 10xx series like mine or @ZeroCool22's is _Pascal_ based, and does not support B-frame encoding in HEVC. You will see the warning message in the log "`Max B frames are 0 frames.`" if you set `bframes` to a positive number. (The default is 3, so the warning always pops up if I forget to set it back to 0. 😢.)

That said, there's one downside about those two recent architectures: **Turing and Ampere are almost always _significantly slower_ than Pascal in HEVC encoding**. I think this is the cost they have to pay to achieve better encoding quality with B-frames and other good features.

But in terms of quality, a lot of users report that their HEVC quality is definitely better than Pascal's and comparable to x265 in many cases.

BTW, the only difference between Ampere(RTX 30 series) and Turing(RTX 20 series, etc.) lies in AV1 decoding. There's almost no benefit of buying an RTX 30 series over an RTX 20 series model in terms of encoding.

**Reply by Dendraspis** [reply]
@ZeroCool22 
What CPU do you have?

@JJKylee 
Thanks a lot! Although my GPU is a *Turing*, I think I stick to x265. 😊

**Reply by ZeroCool22** [reply]
> 
> 
> @ZeroCool22
> What CPU do you have?

i7 7700k.

**Reply by ZeroCool22** [reply]
> 
> 
> There's no such 'encoder' as H265. You're talking about _NVEnc_ **encoder** for _H265(HEVC)_ **codec(=coding format)**. You need to distinguish them in order to avoid confusion.
> 
> BTW, this is my favorite setting for an 8-bit HEVC encoding.
> 
> ![StaxRip_2 1 7 0_NVEnc_settings_8bits_20210110](https://user-images.githubusercontent.com/3157518/104130021-6f3a8900-5323-11eb-96a5-59ac160d3884.jpg)
> 
> Hardware multipass enabled, and the maximum quality preset in the constant quality mode.
> 
> And about the bit depth:
> I don't know how strong the internal deblocking algorithm operates for NVEnc, so instead of adopting 10 bits, I go with 8 bits for an 8-bit source along with the hardware deband filter `--vpp-deband`, _unlike_ `x265` or `aomenc`.
> Maybe you can go with 10 bits instead suited to your taste.
> 
> The remaining options are just pretty self explanatory. 😄

I don¡'t see that Multipass Option...
![Screenshot_5](https://user-images.githubusercontent.com/13344308/104148225-31d00d00-53b0-11eb-891b-8f56552b3c1e.png)

**Reply by JJKylee** [reply]
Multipass mode is enabled only for `--vbr` and `--cbr`. [API v10.0]
See the [NVEnc options manual](https://github.com/rigaya/NVEnc/blob/master/NVEncC_Options.en.md).

**Reply by ZeroCool22** [reply]
> 
> 
> Multipass mode is enabled only for `--vbr` and `--cbr`. [API v10.0]
> See the [NVEnc options manual](https://github.com/rigaya/NVEnc/blob/master/NVEncC_Options.en.md).

Alright! 

So VBR with Multipass is better than **VBR HQ**?

You downloaded the file i leave in the comment?

**Reply by Dendraspis** [reply]
> > @ZeroCool22
> > What CPU do you have?
> 
> i7 7700k.

Funny, I encode with an i7-4770k.  😂 ...and with slow/quality settings. 😎

**Reply by ZeroCool22** [reply]
Ok, but how much hours it takes you?

**Reply by JJKylee** [reply]
> So VBR with Multipass is better than **VBR HQ**?

Of course it **depends on the source video**. There's **no universal solution** in encoding.
BTW, VBR HQ is the RC mode disposed of in NVENC API 10.0. I don't know how rigaya keeps it, but my guess is that he's running it in legacy mode.

> You downloaded the file i leave in the comment?

Nope.
I think it is totally _subjective_ and mostly _depends on the source_ to determine which settings are good or bad for a specific source: except for some key options, of course.
For this reason, I'm willing to talk about key options in general, but not to give recommendations on specific cases unless there are errors.

**Reply by Dendraspis** [reply]
> > > > @ZeroCool22
> > > > What CPU do you have?
> > > 
> > > 
> > > i7 7700k.
> > 
> > 
> > Funny, I encode with an i7-4770k. 😂 ...and with slow/quality settings. 😎
> 
> Ok, but how much hours it takes you?

Totally depends on the source. Most 1080p encodes take about 12-14 hours, depending what I'm doing while it converts. 😊

2160p encodes take about 60 hours.  😕

**Reply by ZeroCool22** [reply]
OMFG!

I guess you use that PC just for encoding right?

**Reply by Dendraspis** [reply]
No, it's my main PC where I do almost everything - also StaxRip coding. 😊

**Reply by ZeroCool22** [reply]
> 
> 
> > So VBR with Multipass is better than **VBR HQ**?
> 
> Of course it **depends on the source video**. There's **no universal solution** in encoding.
> BTW, VBR HQ is the RC mode disposed of in NVENC API 10.0. I don't know how rigaya keeps it, but my guess is that he's running it in legacy mode.
> 
> > You downloaded the file i leave in the comment?
> 
> Nope.
> I think it is totally _subjective_ and mostly _depends on the source_ to determine which settings are good or bad for a specific source: except for some key options, of course.
> For this reason, I'm willing to talk about key options in general, but not to give recommendations on specific cases unless there are errors.

I always do my Rips from a **REMUX** source, i have been told it's the best option, because BDRips and BRRips already have a big compression done.

Multipass will increase the coding time right?

BTW, thx for you response!

**Reply by ZeroCool22** [reply]
> 
> 
> There's no such 'encoder' as H265. You're talking about _NVEnc_ **encoder** for _H265(HEVC)_ **codec(=coding format)**. You need to distinguish them in order to avoid confusion.
> 
> BTW, this is my favorite setting for an 8-bit HEVC encoding.
> 
> ![StaxRip_2 1 7 0_NVEnc_settings_8bits_20210110](https://user-images.githubusercontent.com/3157518/104130021-6f3a8900-5323-11eb-96a5-59ac160d3884.jpg)
> 
> Hardware multipass enabled, and the maximum quality preset in the constant quality mode.
> 
> And about the bit depth:
> I don't know how strong the internal deblocking algorithm operates for NVEnc, so instead of adopting 10 bits, I go with 8 bits for an 8-bit source along with the hardware deband filter `--vpp-deband`, _unlike_ `x265` or `aomenc`.
> Maybe you can go with 10 bits instead suited to your taste.
> 
> The remaining options are just pretty self explanatory. 😄

Will try your settings and compare the result.

This are the correct settings for **Deband** right?:

![Screenshot_6](https://user-images.githubusercontent.com/13344308/104152017-34852f00-53bd-11eb-903c-febb687b723c.png)

**Reply by Dendraspis** [reply]
Remux only says that the streams are not changed/compressed after taken from the original disk. But the quality and sort of a video can be totally different. For example grain and resolution of the origin. A grainy video must be encoded differently than a flat one.

**Reply by JJKylee** [reply]
> This are the correct settings for **Deband** right?:
> 
> ![Screenshot_6](https://user-images.githubusercontent.com/13344308/104152017-34852f00-53bd-11eb-903c-febb687b723c.png)

Yes, but like many other options, the values should be chosen _based on the source itself_. You need to look up the manual and do some trial and error in this regard.

Like @Dendraspis mentioned, _Remux_ is a simple rip of the BD image (iso, etc.) that does _not_ do re-encoding. That is, it just copies streams (video, audio, subtitles, chapters, etc.) from the original source and puts them in a new container like `mkv`. So the quality is preserved as the original source.

But if the source itself is problematic - noise, grain, color anomaly, geometric anomaly, letterboxes, etc. - you need to fix it in encoding. This is where StaxRip comes into play.

BTW, you need to pay full attention to the source to check if it's interlaced or telecined. This is esp. true when you deal with DVD/BD sources. Deinterlacing/IVTC-ing can be done easily in StaxRip as you know, and if you adopt NVEnc, you have two options: software filters of AVS/VS or hardware filters embedded in NVEnc. The decision is totally up to you, but usually hardware filters are faster with some loss in quality.

**Comment by ZeroCool22** [comment]
Finally, combining my settings with the settings you told, i' m very happy with the results.
